src/k8s/pkg/k8sd/database/feature_status.go

```diff
@@ -0,0 +1,79 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/microcluster/cluster"
+)
+
+var featureStatusStmts = struct {
+	select_ int
+	upsert_ int
+}{
+	select_: MustPrepareStatement("feature-status", "select.sql"),
+	upsert_: MustPrepareStatement("feature-status", "upsert.sql"),
+}
+
+// SetFeatureStatus updates the status of the given feature.
+func SetFeatureStatus(ctx context.Context, tx *sql.Tx, name string, status types.FeatureStatus) error {
+	upsertTxStmt, err := cluster.Stmt(tx, featureStatusStmts.upsert_)
+	if err != nil {
+		return fmt.Errorf("failed to prepare upsert statement: %w", err)
+	}
+
+	if _, err := upsertTxStmt.ExecContext(ctx,
+		name,
+		status.Message,
+		status.Version,
+		status.UpdatedAt.Format(time.RFC3339),
```

Comment:
i think it should be possible to pass (and retrieve) `time.Time` objects directly

Reply:
looks like it fails with
`failed to scan row: sql: Scan error on column index 3, name "timestamp": unsupported Scan, storing driver.Value type string into type *time.Time`